ApollosProject/apollos-apps
# Adds Interact when Loaded Component

## DESCRIPTION

### What does this PR do, or why is it needed?

![2020-03-18 10 50 25](https://user-images.githubusercontent.com/1637694/76974495-ab6daa80-6907-11ea-8518-981732c98b11.gif)



Companion PR https://github.com/ApollosProject/apollos-templates/pull/14

This PR adds a component that, when mounted and has a `loading` prop that's `true`, fires an interaction. You can define what you want the interaction to be, and what the nodeId should be. Both of those props are required. 

### What design trade-offs/decisions were made?

Chose a simplistic "completed when opened" direction. Down the line we can come up with more complex ways to define the completion metric. 

### How do I test this PR?

1. Pull the companion PR in `apollos-templates`. 
2. Clone this PR.
3. Make sure the `templates` and `apps` repo are linked. 
4. Run the app. 
5. Clear your server console
6. Click on any content item.
7. Look for the interactionMutation in the server console. Should look something like this.

```
apollos-church-api: While running query: interactWithNode
apollos-church-api:       Total Network Calls: 5
apollos-church-api:       People/GetCurrentPerson: 1
apollos-church-api: /Interactions: 1
apollos-church-api: EntityTypes?%24filter=Name eq 'Rock.Model.ContentChannelItem'&%24top=1: 1
apollos-church-api: InteractionChannels?%24filter=(Name eq 'Apollos App - Content Channel Item') and (UsesSession eq false) and (IsActive eq true) and (ComponentEntityTypeId eq 208) and (ChannelTypeMediumValueId eq 512)&%24top=1: 1
```
 

### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
